app/models/user.rb

```diff
@@ -1,7 +1,7 @@
 class User < ApplicationRecord
   validates :first_name, presence: true
   validates :last_name, presence: true
-  validates :email, presence: true
+  validates :email, presence: true, :uniqueness => true
```

Comment:
Rails uniqueness validations are helpful, but they can create a race condition. Imagine two people signing up the same email and hitting the button at exactly the same time. Suppose you had two processes handling the two requests at the same time. They both check to see if the email exists, find that it doesn't, and then they both commit their data, creating duplicate emails in the database.  To support this validation, you can add a uniqueness index in the database. The database then also ensures that uniqueness is maintained.  See https://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/add_index